### ISSUE TITLE
Disassembled output should be in standard form, as emitted by SPIRV-Tools disassembler

### ISSUE THREAD

**Issue by dneto0** [google/shaderc#79]
It almost certainly implies linking to SPIRV-Tools itself.

**Comment by antiagainst** [comment]
An interesting issue related here is duplicate SPIR-V headers. If we link SPIRV-Tools in, then we have duplicate SPIR-V headers. One is `spirv.hpp` in glslang, The other is `spirv.h` in SPIRV-Tools. Sadly, both of these two headers have `#ifndef spirv_H` & `#define spirv_H` inside them.

**Comment by antiagainst** [comment]
refer to https://github.com/KhronosGroup/SPIRV-Tools/issues/24

**Comment by dneto0** [comment]
Until the header guards are made different, the workaround is that if you need both, then you have to include them from separate compilation units.  So you can wrap the calls to the SPIR-V APIs in a different compilation unit in Shaderc.   It's a bit ugly, but certainly doable.

**Comment by antiagainst** [comment]
@dneto0 It turns out that we don't have some source code pulling in these two headers together. (Yay!) But I've something ugly which should work:

```
// after including header 1:
#ifdef spirv_H
#undef spirv_H
#undef SPIRV_VERSION
#undef SPIRV_REVISION
#endif 
// include header 2
```

It's lucky that everything in `spirv.hpp` is in the `spv` namespace. So we just need to `#undef` the two macros.

**Comment by AWoloszyn** [comment]
@antiagainst  That will unfortunately break the header-guards for header2. So if you end up including something else that includes header2 you will still get double-definitions.
#ifndef spirv_H
#inlclude header1
#undef spirv_H
#undef SPIRV_VERSION
#undef SPIRV_REVISION
#include header2
#endif 
Is a BIT more correct, but it fails if something you include #includes header1 or header2 elsewhere.

**Comment by antiagainst** [comment]
@AWoloszyn: yeah. That's better. Luckily we don't need this kind of hack, at least for now. :)

**Comment by antiagainst** [comment]
Addressed via #81.
